app/modules/auth/auth_router.py

```
@@ -52,52 +67,958 @@ async def login(login_request: LoginRequest):
+                    email_verified=body.get("emailVerified", False),
+                    created_at=first_login,
+                    last_login_at=first_login,
+                    provider_info=provider_info,
+                    provider_username=provider_username,
+                )
+                uid, message, error = user_service.create_user(user)
+
+                # Check if user creation failed due to duplicate email
+                if (
+                    error
+                    and "duplicate" in message.lower()
```

Ruff (0.14.8)

Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)